plans/phase-1/api-design/multigres-operator-api-v1alpha1-design.md

```diff
@@ -0,0 +1,1121 @@
+
+The formalized parent/child model addresses these by ensuring:
+
+  * **Separation of Concerns:** Splitting logic into child CRs results in simple, specialized controllers.
+  * **Single Source of Truth:** The `MultigresCluster` is the only editable entry point for cluster topology, preventing conflicting states.
+  * **Scoped Reusability:** By splitting templates into `CoreTemplate`, `CellTemplate`, and `ShardTemplate`, we provide clear, reusable configurations.
+  * **Explicit Topology:** Removing "shard count" partitioning in favor of explicit shard list definitions provides deterministic control over where exactly data lives.
+  * **Consistent Override Chain:** All components follow a predictable 4-level override chain, providing maximum flexibility while maintaining a clear and consistent API pattern.
+
+## Proposal: API Architecture and Resource Topology
+
+  * **Core Components:** `MultiAdmin` and `GlobalTopoServer` are defined as top-level fields in the `MultigresCluster`. Each can be configured inline or by referencing a `CoreTemplate`.
+  * **Cells:** Explicitly defined in the root CR; can be specified inline or via a `CellTemplate`.
+  * **Databases:** Follows a strict physical hierarchy: `Database` -\> `TableGroup` -\> `Shard`.
+  * **Shards:** Can be specified inline or via a `ShardTemplate`.
+  * **MultiOrch:** Located at the **Shard** level to provide dedicated orchestration per Raft group.
+
+```ascii
+[MultigresCluster] 🚀 (Root CR - user-editable)
+      │
+      ├── 📍 Defines [TemplateDefaults] (Cluster-wide default templates)
+      │
+      ├── 🌍 [GlobalTopoServer] (Child CR) ← 📄 Uses [CoreTemplate] OR inline [spec]
+      │
+      ├── 🤖 MultiAdmin Resources ← 📄 Uses [CoreTemplate] OR inline [spec]
+      │
+      ├── 💠 [Cell] (Child CR) ← 📄 Uses [CellTemplate] OR inline [spec]
+      │    │
+      │    ├── 🚪 MultiGateway Resources
+      │    └── 📡 [LocalTopoServer] (Child CR, optional)
+      │
+      └── 🗃️ [TableGroup] (Child CR)
+           │
+           └── 📦 [Shard] (Child CR) ← 📄 Uses [ShardTemplate] OR inline [spec]
+                │
+                ├── 🧠 MultiOrch Resources (Deployment/Pod)
+                └── 🏊 Pools (StatefulSets for Postgres+MultiPooler)
+
+📄 [CoreTemplate] (User-editable, scoped config)
+   ├── globalTopoServer
+   └── multiadmin
+
+📄 [CellTemplate] (User-editable, scoped config)
+   ├── multigateway
+   └── localTopoServer (optional)
+
+📄 [ShardTemplate] (User-editable, scoped config)
+   ├── multiorch
+   └── pools (postgres + multipooler)
+```
+
+## Design Details: API Specification
+
+### User Managed CR: MultigresCluster
+
+  * This CR and the three scoped templates (`CoreTemplate`, `CellTemplate`, `ShardTemplate`) are the *only* editable entries for the end-user.
+  * All other child CRs will be owned by this top-level CR. Any manual changes to those child CRs will be automatically reverted by their respective controllers to match the `MultigresCluster` definition.
```

Comment:
The rollback won't be done by the respective controllers, as those controllers will be checking for more targeted resource and provision Pods, etc. It is `MultigresCluster` controller who will revert any manual changes made to the child resources.